src/ol_infrastructure/components/aws/database.py

```diff
@@ -52,7 +85,100 @@ def is_valid_version(cls: 'OLDBConfig', engine_version: Text, values: Dict) -> T
         return engine_version
 
 
-class AmazonPostgresDB(pulumi.ComponentResource):
+class OLPostgresDBConfig(OLDBConfig):
+    engine: Text = 'postgres'
+    engine_version: Text = '12.3'
+    port: PositiveInt = PositiveInt(5432)
+    parameter_overrides: List[Dict[Text, Union[Text, bool, int, float]]] = [
+        {'name': 'client_encoding', 'value': 'UTF-8'},
+        {'name': 'server_encoding', 'value': 'UTF-8'},
+        {'name': 'log_timezone', 'value': 'UTC'},
+        {'name': 'timezone', 'value': 'UTC'},
+        {'name': 'rds.force_ssl', 'value': 1}
+    ]
+
+
+class OLMariaDBConfig(OLDBConfig):
+    engine: Text = 'mariadb'
+    engine_version: Text = '10.4.8'
+    port: PositiveInt = PositiveInt(3306)
+    parameter_overrides: List[Dict[Text, Union[Text, bool, int, float]]] = [
+        {'name': 'character_set_client', 'value': 'utf8mb4'},
+        {'name': 'character_set_connection', 'value': 'utf8mb4'},
+        {'name': 'character_set_database', 'value': 'utf8mb4'},
+        {'name': 'character_set_filesystem', 'value': 'utf8mb4'},
+        {'name': 'character_set_results', 'value': 'utf8mb4'},
+        {'name': 'character_set_server', 'value': 'utf8mb4'},
+        {'name': 'time_zone', 'value': 'UTC'}
+    ]
+
+
+class OLAmazonDB(pulumi.ComponentResource):
 
     def __init__(self, db_config: OLDBConfig, opts: pulumi.ResourceOptions = None):
-        super().__init__('ol:infrastructure:AmazonPostgresDB', db_config.db_name, None, opts)
+        super().__init__('ol:infrastructure:aws:rds:PostgresDB', db_config.instance_name, None, opts)
```

Comment:
Where is `PostgresDB` defined?

Reply:
It's not, this is just an arbitrary string but thanks for calling that out. Changing to reflect the generic use of this class.